eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -121,89 +121,10 @@ jobs:
       vmImage: "$(OSVmImage)"
 
     steps:
-      - template: ../steps/common.yml
-
-      - task: PublishPipelineArtifact@1
-        condition: succeededOrFailed()
-        displayName: "Publish Report Artifacts"
-        inputs:
-          artifactName: package-diffs
-          path: $(Build.ArtifactStagingDirectory)
-
-      - script: |
-          pip install setuptools wheel
-          pip install doc-warden==$(DocWardenVersion)
-          ward scan -d $(Build.SourcesDirectory) -c $(Build.SourcesDirectory)/eng/.docsettings.yml
-        displayName: "Verify Readmes"
-
-      - template: /eng/common/pipelines/templates/steps/verify-path-length.yml
-        parameters:
-          SourceDirectory: $(Build.SourcesDirectory)
-
-      - template: ../../../common/pipelines/templates/steps/verify-links.yml
+      - template: ../steps/analyze.yml
         parameters:
-          Directory: sdk/${{ parameters.ServiceDirectory }}
-
-      - script: |
-          npm ci
-        workingDirectory: $(System.DefaultWorkingDirectory)/eng/tools/analyze-deps
-        displayName: "Install tool dependencies"
-
-      - script: |
-          node index.js --verbose --dump "$(Build.ArtifactStagingDirectory)/data.js" --out "$(Build.ArtifactStagingDirectory)/dependencies.html"
-        workingDirectory: $(System.DefaultWorkingDirectory)/eng/tools/analyze-deps
-        displayName: "Analyze library dependencies"
-
-      - script: |
-          node common/scripts/install-run-rush.js install
-        displayName: "Install library dependencies"
-
-      - pwsh: |
-          node common/scripts/install-run-rush.js build -t @azure/eslint-plugin-azure-sdk
-          node eng/tools/rush-runner.js lint "${{parameters.ServiceDirectory}}"
-        displayName: "Build ESLint Plugin and Lint Libraries"
```

Comment:
I did a diff locally and I believe this is the only part missing from the new `analyze.yml` file. 

A nitpick that I am not asking you to do in this PR: it would be nice to make the important change in a commit and then doing the refactoring in another so it is easier to review.